compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangParserListener.java

```diff
@@ -866,9 +866,30 @@ public void exitInclusiveRecordTypeDescriptor(BallerinaParser.InclusiveRecordTyp
 
         boolean isAnonymous = !(ctx.parent.parent instanceof BallerinaParser.FiniteTypeUnitContext);
```

Comment:
I think we can test the complete logic here when we set `isAnonymous` itself. This is to prevent setting the `isAnonymous`  again and again. Good call on creating a function. But it is better to move this logic inside to your function. 
Check here also.
https://github.com/ballerina-platform/ballerina-lang/blob/b9f51f056dd4223ecd8b6b77e5c0df5ca70adbfa/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangParserListener.java#L489

Reply:
Yeah I too feel like should be moved to the function itself. I'll fix

Reply:
Fixed in 6188bc7